setup.py

```diff
@@ -456,7 +456,36 @@ def build_dlib():
         """use cmake to build and install the extension
         """
         if cmake_path is None:
-            raise DistutilsSetupError("Cannot find cmake in the path. Please specify its path with --cmake parameter.")
+            cmake_install_url = "https://cmake.org/install/"
+            message = ("You can install cmake using the instructions at " +
+                       cmake_install_url)
+            if sys.platform == "darwin":
+                pkgmanagers = ('brew', 'port')
+                for manager in pkgmanagers:
+                    if find_executable(manager) is not None:
+                        message = ("You can install cmake on OSX using {0}"
+                                   " with the command `{0} install cmake`."
+                                   .format(manager))
+                        break
+            else:
+                distname, _, __ = platform.linux_distribution()
```

Comment:
So, it seems they were in favour of deprecating it because it's implementation needed to be changed freqeuntly https://bugs.python.org/issue1322 and they recommend a pypi package for it because of this.
This is a package made from that code - https://pypi.python.org/pypi/distro

Perhaps we can add it to `setup_requires` and use it ?